src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -197,10 +198,45 @@ public bool IsSetAsOutput
         /// </summary>
         public string Name
         {
-            get { return nodeLogic.Name; }
+            get 
+            {
+                if (OriginalName != nodeLogic.Name)
+                    IsRenamed = true;
+
+                return nodeLogic.Name; 
+            }
             set { nodeLogic.Name = value; }
         }
 
+        /// <summary>
+        /// The original name of the node.
+        /// This is the name of the node as it is when first added to the canvas.
+        /// </summary>
+        public string OriginalName
```

Comment:
Same here, this change will add attributes to Dynamo json schema, it would be best we can avoid serializing new info at this point.

Reply:
do we need this if we have the method ?